File: Edit-Profile-Service/src/main/kotlin/com/kinok0/editprofileservice/service/CommentService.java
package com.kinok0.editprofileservice.service;

import com.kinok0.editprofileservice.entity.CommentEntity;
import com.kinok0.editprofileservice.entity.TaskEntity;
import com.kinok0.editprofileservice.entity.UserEntity;
import com.kinok0.editprofileservice.repository.CommentRepository;
import com.kinok0.editprofileservice.repository.TaskRepository;
import com.kinok0.editprofileservice.repository.UserRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
public class CommentService {

    private CommentRepository commentRepository;
    private UserRepository userRepository;
    private TaskRepository taskRepository;

    public CommentService(CommentRepository commentRepository, UserRepository userRepository, TaskRepository taskRepository) {
        this.commentRepository = commentRepository;
        this.userRepository = userRepository;
        this.taskRepository = taskRepository;
    }

    public List<CommentEntity> findAll() {
        return commentRepository.findAll();
    }


}

File: Profile-Service/src/main/kotlin/com/kinok0/profileservice/service/CommentService.kt
package com.kinok0.profileservice.service

import com.kinok0.profileservice.dto.Task
import com.kinok0.profileservice.dto.request.CommentRequest
import com.kinok0.profileservice.entity.CommentEntity
import com.kinok0.profileservice.entity.TaskEntity
import com.kinok0.profileservice.repository.CommentRepository
import com.kinok0.profileservice.repository.TaskRepository
import com.kinok0.profileservice.repository.UserRepository
import com.kinok0.profileservice.util.convertToTaskDTO
import org.springframework.stereotype.Service

@Service
class CommentService(
    private val userRepository: UserRepository,
    private val taskRepository: TaskRepository,
    private val jwtService: JwtService,
    private val commentRepository: CommentRepository
) {

    fun createComment(token: String, request: CommentRequest): Task {
        val comment = CommentEntity(
            comment = request.text,
            user = userRepository.getUserEntityByLogin(jwtService.extractUsername(token)),
            task = taskRepository.getTaskEntityById(request.taskId)
        )
        commentRepository.save(comment)
        return taskRepository.getTaskEntityById(request.taskId).convertToTaskDTO()
    }

}